### PR TITLE
[SYCLomatic] Enable migration of function __trap

### PR DESCRIPTION
**Pull request by intwanghao**

### CODE REVIEW

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 1]
```diff
@@ -351,6 +351,10 @@ DPCT_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::bits<DPCPPExtensionsDefaultEnabl
              DPCT_OPT_ENUM("peer_access",
                                      int(DPCPPExtensionsDefaultEnabled::ExtDE_PeerAccess),
                                      "Use the peer access extension.",
+                                     false),
+             DPCT_OPT_ENUM("assert",
```
Comment (ziranzha) [inline clang/include/clang/DPCT/DPCTOptions.inc:355]: Do we need this option? Or just replace with "assert" whatever?

Comment (zhimingwang36) [reply clang/include/clang/DPCT/DPCTOptions.inc:355]: Yes, for all ext,  we need an option to control

File: clang/lib/DPCT/Diagnostics.inc [hunk 2]
```diff
@@ -274,6 +274,8 @@ DEF_WARNING(VEC_IN_TEMPLATE_ARG, 1122, LOW_LEVEL, "'{0}' is migrated to '{1}' in
 DEF_COMMENT(VEC_IN_TEMPLATE_ARG, 1122, LOW_LEVEL, "'{0}' is migrated to '{1}' in template declare, it may cause template function or class redefinition, please adjust the code.")
 DEF_WARNING(UNDEDUCED_KERNEL_FUNCTION_POINTER, 1123, MEDIUM_LEVEL, "The kernel function pointer cannot be used in the device code. You need call the kernel function with correct argunemt(s) directly. According to the kernel function definition, adjusting the dimension of the sycl::nd_item may be also required.")
 DEF_COMMENT(UNDEDUCED_KERNEL_FUNCTION_POINTER, 1123, MEDIUM_LEVEL, "The kernel function pointer cannot be used in the device code. You need call the kernel function with correct argunemt(s) directly. According to the kernel function definition, adjusting the dimension of the sycl::nd_item may be also required.")
+DEF_WARNING(EXPLICIT_EXTENSION_DISABLE, 1124, LOW_LEVEL, "Migration of '%0' is not supported if '%1' extension is disabled. You can migrate the code with '%1' extension by not specifying --no-dpcpp-extensions=%1.")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Diagnostics.inc]: can we try to reuse 1007?

**Review by zhiweij1 — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**